clients/include/utility.hpp

```diff
@@ -411,6 +411,21 @@ int gen_2d_laplacian(int                  ndim,
     return n;
 }
 
+/* ============================================================================================ */
+/*! \brief  Generate a random dense matrix */
+template <typename T>
+void gen_dense_matrix(int m, int n, T* A, int lda)
```

Comment:
Not that it matters here but this generates a random dense matrix containing only 0's and 1' correct? It might be a better interface if you passed in what the min and max values can be for each entry in the matrix. For example:

for(int j = 0; j < n; ++j)
{
    for(int i = 0; i < m; ++i)
    {
        A[j * lda + i] = rand()%(max-min)) + min;
    }
}

This makes it more general and better matches the function name I think? You can have min and max have default parameters still. Just a suggestion though.

Reply:
I like it.  

What I ultimately want is to control the sparsity %.
Here the idea was, well I want 30% of non-zero coefficients, that's why I roughly wrote rand()%10<7 ? 0 : 1.
Values didn't matter in my case but this is good to pay attention. I want to be able to control the sparsity.

Reply:
Refactored as a dense random sparsity pattern with a sparsity ratio, values are positive and normalized.